File: 两数之和.go
func twoSum(nums []int, target int) []int {
    valueStore := make(map[int]int)
    var result []int
    for i:= 0; i < len(nums); i++ {
        if value, ok := valueStore[target - nums[i]]; ok {
            result = append(result, value, i)
            return result
        }
        valueStore[nums[i]] = i
    }
    return result
}

File: 两数之和.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> mp;
        int n = nums.size();
        for(int i = 0; i < n; ++i) {
            if(mp.count(nums[i]) != 0) {
                return {mp[nums[i]], i};
            }
            mp[target - nums[i]] = i;
        }
        return {};
    }
};